src/WebJobs.Script.WebHost/Security/Authorization/AuthUtility.cs

```diff
@@ -35,21 +36,26 @@ public static bool PrincipalHasAuthLevelClaim(ClaimsPrincipal principal, Authori
                 return true;
             }
 
-            AuthorizationLevel level = AuthorizationLevel.Anonymous;
-            if (Enum.TryParse(principal.FindFirstValue(SecurityConstants.AuthLevelClaimType), out level))
+            var authLevelsOnRequest = principal.FindAll(SecurityConstants.AuthLevelClaimType).Select(claim => Enum.Parse(typeof(AuthorizationLevel), claim.Value));
```

Comment:
>authLevelsOnRequest [](start = 16, length = 19)

Again, you have to be careful about using `IEnumerable<T>`. Every call to `authLevelsOnRequest.Contains` is going to re-evaluate the enumerable, which is an unnecessary performance hit.

Reply:
I definitely agree here. I think I just missed the ToList() here, and that probably indicates I should specify the type as well...

---
In reply to: [181942204](https://github.com/Azure/azure-functions-host/pull/2684#discussion_r181942204) [](ancestors = 181942204)